app/javascript/components/lease_ups/LeaseUpApplicationsPage.js

```diff
@@ -1,39 +1,81 @@
 import React from 'react'
-import { flow } from 'lodash'
+import { flow, isEmpty } from 'lodash'
 
 import LeaseUpApplicationsTableContainer from './LeaseUpApplicationsTableContainer'
 import TableLayout from '../layouts/TableLayout'
-import mapProps from '~/utils/mapProps'
 import { mapListing, mapApplicationPreference } from '~/components/mappers/soqlToDomain'
 import { buildLeaseUpModel } from './leaseUpModel'
 import appPaths from '~/utils/appPaths'
+import apiService from '~/apiService'
+import EagerPagination from '~/utils/EagerPagination'
+import Context from './context'
+import mapProps from '~/utils/mapProps'
+
+class LeaseUpApplicationsPage extends React.Component {
+  state = { loading: false, applications: [], pages: 0 }
+
+  constructor (props) {
+    super(props)
+    this.eagerPagination = new EagerPagination(20, 100)
```

Comment:
Adding a quick note here that we might want to extract 20 and 100 to constants (probably in a reusable file)